pinot-core/src/main/java/com/linkedin/pinot/core/query/pruner/TimeSegmentPruner.java

```diff
@@ -35,19 +38,31 @@
  *
  */
 public class TimeSegmentPruner implements SegmentPruner {
+  private static final Logger LOGGER = LoggerFactory.getLogger(TimeSegmentPruner.class);
 
   @Override
   public boolean prune(IndexSegment segment, BrokerRequest brokerRequest) {
-    SegmentMetadata metadata = segment.getSegmentMetadata();
-    String timeColumn = metadata.getTimeColumn();
+    try {
+      SegmentMetadata metadata = segment.getSegmentMetadata();
+      String timeColumn = metadata.getTimeColumn();
+      if (timeColumn == null) {
+        return false;
+      }
 
-    if (timeColumn == null) {
+      FieldSpec.DataType columnDataType = metadata.getColumnDataType(timeColumn);
+      if (columnDataType != FieldSpec.DataType.INT && columnDataType != FieldSpec.DataType.LONG) {
+        return false;
+      }
+
+      TimeInterval segmentInterval = new TimeInterval(metadata.getStartTime(), metadata.getEndTime());
+      FilterQueryTree filterQueryTree = RequestUtils.generateFilterQueryTree(brokerRequest);
+
+      return (filterQueryTree != null && pruneSegment(filterQueryTree, segmentInterval, timeColumn));
+    } catch (Exception e) {
+      // If anything goes wrong during pruning, just return false to be safe.
+      LOGGER.error("Unexpected exception during pruning segment '{}' on time.", segment.getSegmentName(), e);
```

Comment:
this may be very noisy printing for every non-numeric-timestamp segment of every query